lib/piper/permissions/ast/chars/rule.ex

```diff
@@ -3,7 +3,12 @@ defimpl String.Chars, for: Piper.Permissions.Ast.Rule do
   alias Piper.Permissions.Ast
 
   def to_string(%Ast.Rule{command_selector: cs, permission_selector: ps}) do
-    "when #{cs} must have #{ps}"
+    "when #{cs} #{ps_str(ps)}"
   end
 
+  defp ps_str(%Ast.ConditionalExpr{op: :allow}=ps),
+    do: "#{ps}"
+  defp ps_str(ps),
+    do: "must have #{ps}"
```

Comment:
Test with piper/master:
```elixir
iex(1)> {:ok, ast, _} = Piper.Permissions.Parser.parse("when command is foo:bar must have foo:baz")
{:ok, %Piper.Permissions.Ast.Rule{.....}
iex(2)> "#{ast}"
"when command is foo:bar must have foo:baz"
```

If this works now why do we need the function head on line 11?